torch_np/_reductions.py

```diff
@@ -326,13 +261,13 @@ def average_weights(a, axis, w, keepdims=False):
         w = w.swapaxes(-1, axis)
 
     # do the work
-    numerator = torch.mul(a, w).sum(axis)
-    denominator = w.sum(axis)
+    numerator = torch.mul(a, w).sum(axis, dtype=result_dtype)
+    denominator = w.sum(axis, dtype=result_dtype)
     result = numerator / denominator
 
     # keepdims
     if keepdims:
-        result = _util.apply_keepdims(result, ax, ndim)
+        result = _util.apply_keepdims(result, axis, a.ndim)
```

Comment:
IIRC this was problematic in of earlier iterations. As long as CI is green, it's OK now I suppose.